app/src/main/java/org/mozilla/fenix/components/metrics/AppStartupTelemetry.kt

```diff
@@ -143,4 +154,16 @@ class AppStartupTelemetry(private val metrics: MetricController) : LifecycleObse
         // Therefore we set the isMetricRecorded flag to false.
         isMetricRecordedSinceAppWasForegrounded = false
     }
+
+    /**
+     * We record metric on onPreDraw() since we need to wait for the application to draw the
+     * first frame. which will always happen after onResume. The hot startup time is a bit complicated
+     * because if we have onPreDraw listener attached, Activity will invoke onPreDraw() before onRestart.
+     * For this reason, we need to make sure that we remove the preDraw listener when activity is going
```

Comment:
afaict, we never do this: is that a problem?

Reply:
ahh another good reason to avoid comments :sweat_smile: I updated the listener and forgot about the comment. my apologies haha